@here/olp-sdk-core/lib/utils/MultiPartUploadWrapper.ts

```diff
@@ -177,9 +221,211 @@ export class MultiPartUploadWrapper {
+                );
+            }
+
+            await Promise.all(promises);
+        }
+
+        await request.completeMultipartUpload({
+            parts,
+            url: startMultipartResponse.completeUrl,
+            multipartToken: startMultipartResponse.multipartToken,
+            layerId: opts.layerId,
+            billingTag: opts.billingTag
+        });
+
+        blobData.finally && blobData.finally();
+
         return STATUS_CODES.NO_CONTENT;
     }
+
+    private isStartMultipartResponseValid(obj: {
+        multipartToken?: string;
+        uploadPartUrl?: string;
+        completeUrl?: string;
+        statusUrl?: string;
+    }): boolean {
+        return (
+            obj.multipartToken !== undefined ||
+            (obj.uploadPartUrl !== undefined && obj.completeUrl !== undefined)
+        );
+    }
+
+    private isBlobData(input: any): boolean {
+        return (
+            typeof input.size === "function" &&
+            typeof input.readBytes === "function"
+        );
+    }
+
+    private async prepareBlobData(
+        input: string | File | Blob | ArrayBufferLike | BlobData
+    ): Promise<BlobData> {
+        if (this.isBlobData(input)) {
+            return input as BlobData;
+        }
+
+        if (typeof Buffer !== "undefined") {
+            // node.js
+            if (Buffer.isBuffer(input)) {
+                return new BufferData(input);
+            }
+
+            if (typeof input === "string") {
+                return NodeFileData.fromPath(input);
+            }
+        }
+
+        // browser
+        if (typeof input === "string") {
+            throw new Error("Unsupported input");
+        }
+
+        return input instanceof Blob
+            ? new WebData(input)
+            : new BufferData(input as ArrayBuffer);
+    }
+
+    private validateChunkSize(size = 0): number {
+        const minChunkSizeMiB = 5;
+        const mmxChunkSizeMiB = 5120;
+        const bytesInMiB = 1048576;
+        return size >= minChunkSizeMiB && size <= mmxChunkSizeMiB
+            ? size * bytesInMiB
+            : minChunkSizeMiB * bytesInMiB;
+    }
+
+    private async getUploadRequest(opts: {
+        blobVersion: number;
+        catalogHrn: string;
+        abotrSignal?: AbortSignal;
+    }): Promise<UploadRequest> {
+        const cacheKey = `request::blob::v${opts.blobVersion}::catalog::${opts.catalogHrn}`;
+        const cachedRequest = this.cachedUploadRequests.get(cacheKey);
+        if (cachedRequest) {
+            return cachedRequest;
+        }
+
+        const supportedBlobVersions: { [key: string]: boolean } = {
+            v1: true,
+            v2: true
+        };
+
+        if (!supportedBlobVersions[`v${opts.blobVersion}`]) {
+            return Promise.reject(
+                new Error(`Unsupported Blob version: ${opts.blobVersion}`)
+            );
+        }
+
+        const requestBuilder = await RequestFactory.create(
+            "blob",
+            `v${opts.blobVersion}`,
```

Comment:
Does it not make much more sense than to make the opts.blobVersion a string instead and extend documentation so that the user provides "v1" or "v2".